pkg/controller/bucket/bucket_controller.go

```diff
@@ -143,27 +160,43 @@ func (bl *bucketListener) Delete(ctx context.Context, obj *v1alpha1.Bucket) erro
 	}
 
 	req := osspec.ProvisionerDeleteBucketRequest{
-		BucketName: obj.Spec.BucketRequest.Name,
+		BucketContext: map[string]string{},
 	}
 
 	switch obj.Spec.Protocol.Name {
 	case v1alpha1.ProtocolNameS3:
+		req.BucketName = obj.Spec.Protocol.S3.BucketName
 		req.Region = obj.Spec.Protocol.S3.Region
+		req.BucketContext["Version"] = obj.Spec.Protocol.S3.Version
+		req.BucketContext["SignatureVersion"] = string(obj.Spec.Protocol.S3.SignatureVersion)
+		req.BucketContext["Endpoint"] = obj.Spec.Protocol.S3.Endpoint
 	case v1alpha1.ProtocolNameAzure:
+		req.BucketName = obj.Spec.Protocol.AzureBlob.ContainerName
+		req.BucketContext["StorageAccount"] = obj.Spec.Protocol.AzureBlob.StorageAccount
 	case v1alpha1.ProtocolNameGCS:
+		req.BucketName = obj.Spec.Protocol.GCS.BucketName
+		req.BucketContext["ServiceAccount"] = obj.Spec.Protocol.GCS.ServiceAccount
+		req.BucketContext["PrivateKeyName"] = obj.Spec.Protocol.GCS.PrivateKeyName
+		req.BucketContext["ProjectID"] = obj.Spec.Protocol.GCS.ProjectID
 	default:
-		errStr := fmt.Sprintf("unknown protocol: %s", obj.Spec.Protocol.Name)
-		klog.Errorf(errStr)
-		return fmt.Errorf(errStr)
+		return fmt.Errorf("unknown protocol: %s", obj.Spec.Protocol.Name)
 	}
 
 	// TODO set grpc timeout
 	rsp, err := bl.provisionerClient.ProvisionerDeleteBucket(ctx, &req)
 	if err != nil {
 		klog.Errorf("error calling ProvisionerDeleteBucket: %v", err)
+		obj.Status.Message = "Bucket Deleting"
+		obj.Status.BucketAvailable = false
+		_, err = bl.bucketClient.ObjectstorageV1alpha1().Buckets().UpdateStatus(ctx, obj, metav1.UpdateOptions{})
```

Comment:
Should we do these backend calls with retry? 

https://github.com/kubernetes/client-go/blob/master/util/retry/util.go#L48

Reply:
Done

Reply:
I like this idea but isn't the central controller already retrying the delete event?